test/system/shoot_cp_migration/migrate_test.go

```diff
@@ -0,0 +1,255 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+	Overview
+		- Tests the migration of a shoot by using the "framework/shootmigrationtest.go"
+		- Performs sanity checks on the migrated shoot to verify that migration is working as expected.
+ **/
+
+package cp_migration_test
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"time"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	. "github.com/gardener/gardener/test/framework"
+	"github.com/gardener/gardener/test/framework/applications"
+	"github.com/onsi/ginkgo"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	CreateAndReconcileTimeout = 2 * time.Hour
+	SecretName                = "test-shoot-migration-secret"
+	SecretNamespace           = "default"
+	ServiceAccountName        = "test-service-account"
+	ServiceAccountNamespace   = "default"
+)
+
+var (
+	targetSeedName *string
+	shootName      *string
+	shootNamespace *string
+	gardenerConfig *GardenerConfig
+)
+
+func init() {
+	gardenerConfig = RegisterGardenerFrameworkFlags()
+	targetSeedName = flag.String("target-seed-name", "", "name of the seed to which the shoot will be migrated")
+	shootName = flag.String("shoot-name", "", "name of the shoot")
+	shootNamespace = flag.String("shoot-namespace", "", "namespace of the shoot")
+}
+
+var _ = ginkgo.Describe("Shoot migration testing", func() {
+	f := NewGardenerFramework(gardenerConfig)
+	t := NewShootMigrationTest(f, &ShootMigrationConfig{
+		ShootName:      *shootName,
+		ShootNamespace: *shootNamespace,
+		TargetSeedName: *targetSeedName,
+	})
+	guestBookApp := applications.GuestBookTest{}
+	testSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      SecretName,
+			Namespace: SecretNamespace,
+		},
+	}
+	testServiceAccount := &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ServiceAccountName,
+			Namespace: ServiceAccountNamespace,
+		}}
+
+	CBeforeEach(func(ctx context.Context) {
+		validateConfig()
```

Comment:
We don't need to call validateConfig in CBeforeEach(). There's a CBeforeSuite() call which can be used.